code/model/Learner.cpp

```diff
@@ -651,6 +764,26 @@ void SubgoalLearner::TryPlanningOnFullTasks (void)
 	hmp_IndexToSequenceState.clear ();
 }
 
+/**
+ **Given the i_CurrentStep, sets the step to the next non-question subgoal
+ **Note this cannot overflow in regular use as last subgoal is
+ ** ensured to be a real subgoal
+**/
+int SubgoalSequenceState::nextSubgoal(){
+  unsigned int currentStep = this->i_CurrentStep;
+  size_t max_step =  this->p_Sequence->dq_Subgoals.size();
```

Comment:
yes, but i do less than, so its a len

Either Way, by how subgoals are constructed, it can't ever get to the edge and call this function, because last is never a question